pkg/paths/path.go

```diff
@@ -31,9 +31,9 @@ func GetAllParams(c buffalo.Context) (*AllPathParams, error) {
 	if err != nil {
 		return nil, errors.E(op, err)
 	}
-	version := c.Param("version")
-	if version == "" {
-		return nil, errors.E(op, "missing version paramater")
+	version, err := DecodeVersion(c.Param("version"))
```

Comment:
can you remind me why we need `DecodeVersion`? and `DecodePath`? it seems like a lot of duplicated code that would be _amazing_ if we could avoid, so I'm wondering if there's any way around it

Reply:
@arschles find the reason here :) https://github.com/gomods/athens/issues/356